controller/space.go

```diff
@@ -415,8 +415,36 @@ func ConvertSpaceFromModel(ctx context.Context, db application.DB, request *goa.
 	relatedCollaboratorList := rest.AbsoluteURL(request, fmt.Sprintf("/api/spaces/%s/collaborators", spaceIDStr))
 	relatedFilterList := rest.AbsoluteURL(request, "/api/filters")
 
+	// fetch all categories links
+	categoriesData := []*app.GenericData{}
+	err := application.Transactional(db, func(appl application.Application) error {
+		categories, err := appl.Categories().List(ctx)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
+			}, "unable to fetch categories")
+			return errs.Wrap(err, "unable to fetch categories")
+		}
+		for _, cat := range categories {
+			relatedCategoryList := rest.AbsoluteURL(request, fmt.Sprintf("/api/spaces/%s/workitems?filter[category]=%s", spaceIDStr, cat.ID))
+			catList := app.GenericData{
+				Links: &app.GenericLinks{
+					Related: &relatedCategoryList,
+				},
+			}
+			categoriesData = append(categoriesData, &catList)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errs.Wrap(err, "unable to fetch categories")
```

Comment:
This error should be wrapped by `return jsonapi.JSONErrorResponse(ctx, err)`. Also you are doing it twice